pkg/v1/layout/read.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2019 The original author or authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package layout
+
+import (
+	"io/ioutil"
+	"path/filepath"
+)
+
+// Read reads an OCI image layout at path and constructs a LayoutPath.
+func Read(path string) (LayoutPath, error) {
+	// TODO: check oci-layout exists
+
+	_, err := ioutil.ReadFile(filepath.Join(path, "index.json"))
```

Comment:
Seems wasteful to read this without doing anything with it. You could just stat the file or stash it away for a later call to `ImageIndex`?

I'd also want some way to open a LayoutPath for writing without having to write an empty index first. Maybe we could change this to `Open` (or add another method) that will initialize to an empty.Index if not found.

Reply:
> Seems wasteful to read this without doing anything with it. You could just stat the file or stash it away for a later call to `ImageIndex`?

Changed to use stat.

> 
> I'd also want some way to open a LayoutPath for writing without having to write an empty index first. Maybe we could change this to `Open` (or add another method) that will initialize to an empty.Index if not found.

Ran out of time today. Need to think about this on Monday, but please feel free to merge this and change as necessary before then.